src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -310,7 +310,7 @@ internal async Task IndexMethodAsyncCore(MethodInfo method, IFunctionIndexCollec
 
             if (TypeUtility.IsAsyncVoid(method))
             {
-                string msg = $"Function '{method.Name}' is async but does not return a Task. Your function may not run correctly.";
+                string msg = $"Function '{Utility.GetFunctionName(method)}' is async but does not return a Task. Your function may not run correctly.";
```

Comment:
Can we introduce a `GetFunctionShortName()` that will return `method.GetShortName()` if there's no attribute? We really should have the class name in all of these, if the `FunctionNameAttribute` isn't there. But we can't change `GetFunctionName()` because it's already being used for generating the `Disabled` setting name.

It's a little weird b/c `GetFunctionShortName()` would be longer than `GetFunctionName()`, but it will be consistent with what we call "ShortName" elsewhere. Make sure it has some comments, and since it's internal, it should be okay.

Reply:
I thought of that, was kind of undecided between which way to go. GetFunctionShortName() sounds right.